test/services/objection.service.spec.unit.ts

```diff
@@ -39,15 +40,15 @@ const mockAttachment = {
     name: "test.doc",
 };
 
-mockGetAttachments.prototype.constructor.mockImplementation(async (companyNumber: string, token: string,
-                                                                   objectionId: string): Promise<Attachment[]> => {
+mockGetAttachments.mockImplementation(async (companyNumber: string, token: string,
```

Comment:
if the mock returns a promise, you can use mockResolvedValue/Once (mockAttachments)
for non async functions you can use mockReturnValue.